.github/workflows/go.yml

```diff
@@ -272,6 +283,96 @@ jobs:
           name: smoketest-${{ matrix.smoke-suite }}-cluster-state
           path: /tmp/cluster-state.zip
 
+  autopilot-smoketest-matrix:
+    name: Release matrix
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - id: set-matrix
+        run: |
+          set -x 
+          go install github.com/k0sproject/version/cmd/k0s_sort@v0.2.2
+          GOBIN="$(go env GOPATH)/bin"
+          matrix="["
+          COMMA=""
+          for i in 1.24.2 1.24.3; do \
```

Comment:
The only problem in building up code in GHA yaml is that it becomes harder to catch things locally, before it gets to GHA. So to re-run a failing matrix test scenario would be something like:
* manually download the base k0s binary (`K0S_PATH`)
* have a compiled k0s (`K0S_UPDATE_FROM_PATH`)
* define the update version (`K0S_UPDATE_TO_VERSION`) 
* `make -C inttest check-ap-ha3x3`

Is there any way to move some of this in scripts to make it a bit easier for devs?

Reply:
I've added `hack/tools/run-autopilot-matrix-tests.sh`, I hope it will help